src/expr/src/relation/mod.rs

```diff
@@ -631,6 +631,178 @@ impl MirRelationExpr {
         self.typ().column_types.len()
     }
 
+    /// Recursively descend two expressions until differences are found.
+    ///
+    /// The resulting sequence contains the first moments at which differences were observed.
+    /// There may be multiple elements in the list as the differences may only be observed
```

Comment:
Add a note in this doc that it is assumed that predicates and join equivalences have been canonicalized?

Reply:
It's not assumed or required, but I do think the method either needs to be better explained or removed (probably the latter, as I only use it for diagnostics while developing). It might be good to have this code somewhere, though, to more clearly present differences in expressions as they evolve.